app/controllers/favorites_controller.rb

```diff
@@ -0,0 +1,10 @@
+class FavoritesController < ApplicationController
+
+  def update
+    pet = Pet.find(params[:pet_id])
+    favorites.add_pet({id: pet.id, info: {name: pet.name, image: pet.image}})
+    session[:favorites] = favorites.contents
+    flash[:notice] = "#{pet.name} has been added to favorites!"
+    redirect_to "/pets/#{params[:pet_id]}"
+  end
```

Comment:
looks up favorited pet in database, adds passes its id, name, and image to the PORO (accessed from the helper), updates the session to include the info that was just added

flash notice will have to be updated to handle pets that were already added